app/scenes/Document/components/CommentThreadItem.tsx

```diff
@@ -117,31 +116,24 @@ function CommentThreadItem({
     );
   };
 
-  const handleSubmit = async (event: React.FormEvent) => {
+  const handleSubmit = action(async (event: React.FormEvent) => {
     event.preventDefault();
 
     try {
       setReadOnly();
-      await comment.save({
-        data,
-      });
+      comment.data = data;
+      await comment.save();
     } catch (error) {
       setEditing();
       toast.error(t("Error updating comment"));
     }
-  };
+  });
 
   const handleCancel = () => {
-    setData(toJS(comment.data));
+    setData(comment.data);
     setReadOnly();
-    setForceRender((i) => ++i);
   };
 
-  React.useEffect(() => {
-    setData(toJS(comment.data));
```

Comment:
With `observable.shallow` there is no need to use `toJS` in this component

Reply:
Looks to be the main cause 👌